### PR TITLE
feat gh-269: Http method to patch a secret

### PR DESCRIPTION
**Pull request by konidev20**
Hey @rajanadar, 

Not sure if this is the right way to fix gh-269. Please review it.

### CODE REVIEW

File: src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public async Task<Secret<CurrentSecretMetadata>> PatchSecretAsync(string path, I
                 }
             };
 
-            return await _polymath.MakeVaultApiRequest<Secret<CurrentSecretMetadata>>(mountPoint ?? _polymath.VaultClientSettings.SecretsEngineMountPoints.KeyValueV2, "/data/" + path.Trim('/'), HttpMethod.Post, requestData).ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs:103]: i think the rest of this method becomes moot with patch. No need to read, upsert etc. So remove all that.

Also, please add a changelog entry for 1.9.0 TBD release as this is a 1.9.0 vault api feature. 

Please add a Readme snippet for this as well if not already.

**Review by rajanadar — CHANGES_REQUESTED** [review]

File: src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs (hunk 1 above)
Comment (konidev20) [reply src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs:103]: Hey @rajanadar, 

I reviewed the API documentation [1]. 

I believe we should just remove the part using existing data for the upsert.  I found that the read is used to capture the `cas` flag with the latest version number which is required if the mounted secret engine config or secret in question has `cas_required` set to true. 

We can take 2 approaches here: 
1. Remove the read and upsert code, introduce a new parameter to the interface and implementation with the version number, and make it a required field to set the `cas` flag. Existing users of the library might face issues while upgrading to newer versions.
2. Leave the read code to set the `cas` flag through VaultSharp and use the `newData` object to send the request as a patch.

References:
[1] https://www.vaultproject.io/api-docs/secret/kv/kv-v2#patch-secret

Comment (konidev20) [reply src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs:103]: We can send another dictionary object called `options` forcing the user of the library to follow the API spec. And by default, we can assume that the `cas_required` is set to false.

File: src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs [hunk 2]
```diff
@@ -69,47 +69,11 @@ public async Task<Secret<CurrentSecretMetadata>> WriteSecretAsync<T>(string path
...
-            {
-                data = currentSecret.Data.Data,
-                options = new Dictionary<string, object>
-                {
-                    {  "cas", metadata.Version }
-                }
-            };
-
-            return await _polymath.MakeVaultApiRequest<Secret<CurrentSecretMetadata>>(mountPoint ?? _polymath.VaultClientSettings.SecretsEngineMountPoints.KeyValueV2, "/data/" + path.Trim('/'), HttpMethod.Post, requestData).ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
+            return await _polymath.MakeVaultApiRequest<Secret<CurrentSecretMetadata>>(mountPoint ?? _polymath.VaultClientSettings.SecretsEngineMountPoints.KeyValueV2, "/data/" + path.Trim('/'), new HttpMethod("PATCH"), patchSecretDataRequest).ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/KeyValue/V2/KeyValueSecretsEngineV2Provider.cs:76]: @konidev20 Can you please explain why the existing logic to check current secret etc. was removed?

**Review by rajanadar — APPROVED** [review]

**Comment by rajanadar** [comment]
thanks @konidev20

**Pull request merged**